src/jit/valuenum.cpp

```diff
@@ -215,14 +221,125 @@ struct DoubleTraits
 {
     static double NaN()
     {
+#if defined(_TARGET_XARCH_)
         unsigned long long bits = 0xFFF8000000000000ull;
-        double             result;
+#elif defined(_TARGET_ARMARCH_)
+        unsigned long long bits = 0x7FF8000000000000ull;
+#else
+#error Unsupported or unset target architecture
+#endif
+        double result;
         static_assert(sizeof(bits) == sizeof(result), "sizeof(unsigned long long) must equal sizeof(double)");
         memcpy(&result, &bits, sizeof(result));
         return result;
     }
 };
 
+// The following is needed since NaN returned by floating point arithmetic operation on ARMARCH
+// will have different binary representation (0x7fc00000 float, 0x7ff8000000000000 double)
+// that NaN on XARCH (0xffc00000 float, 0xfff8000000000000 double).
+// This can cause X86_ARM32 and ARM32 crossgens to produce different binary outputs.
```

Comment:
Nit: Instead of, or in addition to this comment, it would be good to have function headers on the individual methods.